kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -977,6 +1005,26 @@ Literal arrays are also built using this RValue.
        </k>
 ```
 
+While Unions are Aggregate in the MIR, we distinguish between them because the semantics
+are different. For example, field accesses to not access different data, but interpret
+that data as a different type.
+```k
+  syntax Rvalue ::= #evalUnion ( Rvalue )
+
+  rule <k> #evalUnion(rvalueAggregate(aggregateKindAdt( _, _, _, _, someFieldIdx ( FIELD )), ARGS))
```

Comment:
Probably we will always have a single argument? (`ARG .Args`) 
That could make the next rule simpler